frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import * as fuzzy from 'fuzzysearch';
+import { RootState } from '@console/internal/redux';
+import { Button } from '@patternfly/react-core';
+import { Dropdown } from '@console/internal/components/utils';
+import { queryBrowserRunQueries, queryBrowserPatchQuery } from '@console/internal/actions/ui';
+import { getActiveNamespace } from '@console/internal/reducers/ui';
+import { QueryInput } from '@console/internal/components/monitoring/metrics';
+import { QueryObj } from '@console/internal/components/monitoring/query-browser';
+import { metricsQuery, getTopMetricsQueries } from '../queries';
+import './MetricsQueryInput.scss';
+
+const ADD_NEW_QUERY = '#ADD_NEW_QUERY#';
+
+type StateProps = {
+  namespace: string;
+};
+
+type MetricsQueryInputProps = {
+  patchQuery: (patch: QueryObj) => void;
+  runQueries: () => void;
+  namespace: string;
+};
+
+export const MetricsQueryInput: React.FC<MetricsQueryInputProps> = ({
+  patchQuery,
+  runQueries,
+  namespace,
+}) => {
+  const items = metricsQuery;
+  const autocompleteFilter = (text, item) => fuzzy(text, item);
+  const defaultActionItem = [
+    {
+      actionTitle: `Custom Query`,
+      actionKey: ADD_NEW_QUERY,
+    },
+  ];
+
+  const [title, setTitle] = React.useState('Select Query');
+  const [metric, setMetric] = React.useState('');
+  const [showPromQl, setShowPromQl] = React.useState(false);
+  const [isPromQlDisabled, setIsPromQlDisabled] = React.useState(false);
```

Comment:
Actually Show PromQL button is not shown in design mock up when no query is selected. If possible its good to hide otherwise we can set `isPromQlDisabled` default value to `true`

Reply:
It's there `Show PromQL button` in the mockup